### DISCUSSION TITLE
Use a templating system for html pages

### DISCUSSION THREAD

**Discussion by wictorwilen** [pnp/generator-teams#297] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently static html pages are used for rendering the tabs and pages. Some things get's hardcoded in those such as application insights, CSS etc. To update that you need to modify *n* similar html files.

**Describe the solution you'd like**
Implement some kind of templating system so there is a *master* html page

**Describe alternatives you've considered**
PIck one... https://expressjs.com/en/resources/template-engines.html. `express-handlebars` looks like a simple and good option

**Comment by andrewconnell** [comment]
LOVE this idea... maybe may it an option in the generator though instead of forced? If someone isn't familiar with handlebars (or whatever else is chosen), it would require they learn it. Making it optional, even if default=yes, would avoid the forcing function.

Maybe merge with #100?

**Comment by wictorwilen** [comment]
Yes, let's discuss a decent approach. It could also make sense to have this with the more modular build pipeline discussed in the gitter conversation, allowing us to make template plugins?

**Comment by StfBauer** [comment]
The easiest way to get this is to use [gulp-file-include](https://github.com/haoxins/gulp-file-include#readme)

```html
...
<body>
   @@include('./header.html')

   Content

   @@include('./footer.html')
</body>
...

A simple gulp task will then compile this to the final HTML and no one need to learn anything other than writing HTML.

**Comment by wictorwilen** [comment]
Thanks @StfBauer - that would be one way of doing templating.
